cibuildwheel/util.py

```diff
@@ -121,7 +123,42 @@ def __repr__(self):
         return f'{self.__class__.__name__}{self.base_file_path!r})'
 
 
+class Architecture(str, Enum):
+    # mac/linux archs
+    x86_64 = 'x86_64'
+    i686 = 'i686'
+    aarch64 = 'aarch64'
+    ppc64le = 'ppc64le'
+    s390x = 's390x'
+
+    # windows archs
+    x86 = 'x86'
+    AMD64 = 'AMD64'
+
+    @staticmethod
+    def parse_config(config: str, platform: str) -> 'List[Architecture]':
+        result = []
+        for arch_str in re.split(r'[\s,]+', config):
+            if arch_str == 'auto':
+                result += Architecture.auto_archs(platform=platform)
+            else:
+                result.append(Architecture(arch_str))
+        return result
+
+    @staticmethod
+    def auto_archs(platform: str) -> 'List[Architecture]':
+        native_architecture = Architecture(platform_module.machine())
+        result = [native_architecture]
+        if platform == 'linux' and native_architecture == Architecture.x86_64:
+            # x86_64 machines can run i686 docker containers
+            result.append(Architecture.i686)
+        if platform == 'windows' and native_architecture == Architecture.AMD64:
+            result.append(Architecture.x86)
+        return result
+
+
 class BuildOptions(NamedTuple):
+    architectures: List[Architecture]
```

Comment:
Nitpick: these aren't alphabetically sorted, but rather chronologically (when it's needed by `cibuildwheel`) or vaguely by topic or something like that. So let's move it between `build_selector` and `environment`?

Reply:
yep